Tests/NIOHTTP1Tests/ByteBufferUtilsTest.swift

```diff
@@ -0,0 +1,79 @@
+    case err
+}
+
+class ByteBufferUtilsTest: XCTestCase {
+    
+    func testComparators() {
+        var someByteBuffer: ByteBuffer = ByteBufferAllocator.init().buffer(capacity: 16)
+        someByteBuffer.write(string: "fiRSt")
+        XCTAssert(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "first".utf8))
+        XCTAssert(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "fiRSt".utf8))
+        XCTAssert(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "fIrst".utf8))
+        XCTAssertFalse(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "fIrt".utf8))
+        XCTAssertFalse(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "firsta".utf8))
+        XCTAssertFalse(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "afirst".utf8))
+        XCTAssertFalse(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "eiRSt".utf8))
+        XCTAssertFalse(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "fIrso".utf8))
+        XCTAssertFalse(
+            someByteBuffer.readableBytesView.compareCaseInsensitiveASCIIBytes(
+                to: "firot".utf8))
+        
+    }
+
+    private func byteBufferView(string: String) -> ByteBufferView {
+        let byteBufferAllocator = ByteBufferAllocator.init()
+        var buffer = byteBufferAllocator.buffer(capacity: string.lengthOfBytes(using: .utf8))
+        buffer.write(string: string)
+        return buffer.readableBytesView
+    }
+
+    func testTrimming() {
+        XCTAssertEqual(byteBufferView(string: "   first").trimSpaces().map({CChar($0)}), byteBufferView(string: "first").map({CChar($0)}))
+        XCTAssertEqual(byteBufferView(string: "   first  ").trimSpaces().map({CChar($0)}), byteBufferView(string: "first").map({CChar($0)}))
+        XCTAssertEqual(byteBufferView(string: "first  ").trimSpaces().map({CChar($0)}), byteBufferView(string: "first").map({CChar($0)}))
+        XCTAssertEqual(byteBufferView(string: "first").trimSpaces().map({CChar($0)}), byteBufferView(string: "first").map({CChar($0)}))
+        XCTAssertEqual(byteBufferView(string: " \t\t  fi  rst").trimSpaces().map({CChar($0)}), byteBufferView(string: "fi  rst").map({CChar($0)}))
+        XCTAssertEqual(byteBufferView(string: "   firs  t \t ").trimSpaces().map({CChar($0)}), byteBufferView(string: "firs  t").map({CChar($0)}))
+        XCTAssertEqual(byteBufferView(string: "f\t  irst  ").trimSpaces().map({CChar($0)}), byteBufferView(string: "f\t  irst").map({CChar($0)}))
+        XCTAssertEqual(byteBufferView(string: "f i  rs  t").trimSpaces().map({CChar($0)}), byteBufferView(string: "f i  rs  t").map({CChar($0)}))
```

Comment:
Can you add a test case here that tests trimming something that's entirely spaces?